modules/rostests/kmtests/ntos_io/IoFilesystem.c

```diff
@@ -166,6 +166,13 @@ TestAllInformation(VOID)
            "Char past FileName is %x\n",
            FileAllInfo->NameInformation.FileName[NameLength / sizeof(WCHAR)]);
     }
+    else
+    {
+        // Following cases require NameLength, so skip them too.
+        if (FileAllInfo)
```

Comment:
This should be fixed inside QueryFileInfo. The function should guarantee that FileAllInfo is freed in case of a non-success status.

Reply:
Done.